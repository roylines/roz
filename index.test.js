const {handler} = require('.');
const {TelegramClient} = require('messaging-api-telegram');

jest.mock('messaging-api-telegram');
process.env.TELEGRAM_TOKEN = 'TELEGRAM_TOKEN';
process.env.TELEGRAM_USER = 'TELEGRAM_USER';

console.log = jest.fn();
describe('handler', () => {
  const mockTelegram = () => {
    const connect = {sendMessage: jest.fn()};
    connect.sendMessage.mockResolvedValue();
    TelegramClient.connect.mockReturnValue(connect);
  };

  it('should call connect correctly', async () => {
    mockTelegram();
    await handler();
    expect(TelegramClient.connect.mock.calls).toMatchSnapshot();
  });

  it('should return expected', async () => {
    mockTelegram();
    const response = await handler();
    expect(response).toMatchSnapshot();
  });
});
