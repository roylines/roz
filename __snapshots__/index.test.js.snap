// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler should call connect correctly 1`] = `
Array [
  Array [
    "TELEGRAM_TOKEN",
  ],
]
`;

exports[`handler should return expected 1`] = `undefined`;
